### ISSUE TITLE
Different behaviour between webpack and karma-webpack

### ISSUE THREAD

**Issue by JackFrostStudios** [ryanclark/karma-webpack#271]
**I'm submitting a bug report**
**Webpack version:** 2.7.0
**Webpack Karma version**: 2.0.4
**Karma version**: 1.3.0
**Environment**: Windows 10
**Browser:** PhantomJs 2.1.7

**Current behavior:**
When compiling ts test files via karma-webpack receive a number of errors relating to type definitions; for example:
TS2503: Cannot find namespace 'angular'

However when building the same file directly through webpack there are no errors.
**Expected/desired behavior:**
karma-webpack should not throw errors when webpack itself can build the files.

**Config Files**:
Webpack config:
```
let config = {
    entry: null,
    output: null,
    devtool: false,
    resolve: {
        extensions: [".ts", ".js"]
    },
    module: {
        rules: [
            {
                test: /\.ts$/,
                enforce: 'pre',
                loader: 'tslint-loader',
                options: {
                    emitErrors: true,
                    failOnHint: true
                }
            },
            {
                test: /\.ts$/,
                use: [
                    {
                        loader: "awesome-typescript-loader"
                    }
                ]
            },
            {
                test: /\.html$/,
                exclude: ["./src/index.html"],
                loader: "raw-loader"
            },
            {
                test: /\.scss$/,
                use: extractSass.extract({
                    use: [{
                        loader: "css-loader",
                        options: {
                            minimize: true,
                            sourceMap: false,
                            importLoaders: 1
                        }
                    },
                    {
                        loader: "sass-loader",
                        options: {
                            outputStyle: "compressed",
                            sourceMap: false,
                            sourceMapContents: false
                        }
                    }
                    ],
                    fallback: "style-loader"
                })
            },
            {
                test: /\.(eot|svg|ttf|woff|woff2)$/,
                loader: "file-loader",
                include: helpers.root("/src/fonts"), // this is the path to the copied fonts in your project
                options: {
                    regExp: /(fonts\/.*)/,
                    name: "[name].[ext]",
                    publicPath: "../",
                    outputPath: "fonts/"
                }
            },
            {
                test: /\.svg$/,
                exclude: [
                    helpers.root("/src/fonts")
                ],
                use: [
                    {
                        loader: "svg-sprite-loader",
                        options: {
                            esModule: false
                        }
                    },
                    {
                        loader: "svgo-loader",
                        options: {
                            plugins: [
                                { cleanupAttrs: true },
                                { removeDoctype: true },
                                { removeComments: true },
                                { removeMetadata: true },
                                { removeTitle: true }
                            ]
                        }
                    }
                ]
            }
        ],
    },
    plugins: [
        extractSass
    ],
    externals: {
        jquery: { root: '$', amd: 'jquery', commonjs: 'jquery', commonjs2: 'jquery' }
    }
};

module.exports = config;
```

karma config:
```
module.exports = function (config) {
    config.set({
        basePath: "",
        frameworks: ["jasmine", "source-map-support"],
        files: [
            "node_modules/jquery/dist/jquery.js",
            "src/vendor-bundle.ts",
            "node_modules/angular-mocks/angular-mocks.js",
            "node_modules/lodash/lodash.min.js",
            "node_modules/postal/lib/postal.min.js",
            "src/main.ts",
            "src/test.ts"
        ],
        preprocessors: {
            "src/main.ts": ["webpack"],
            "src/vendor-bundle.ts": ["webpack"],
            "src/test.ts": ["webpack"]
        },
         mime: {
            'text/x-typescript': ['ts','tsx']
        },
        webpack: webpackConfig,
        webpackServer: { noInfo: true },
        reporters: ["mocha"],
        port: 9876,
        colors: true,
        logLevel: config.LOG_INFO,
        autoWatch: false,
        browsers: ["PhantomJS"],
        singleRun: true
    });
};
```

Webpack config used when building the test.ts directly in webpack:
```
const webpack = require("webpack"),
    webpackConfig = require("./webpack.config.base"),
    helpers = require("./helpers");

webpackConfig.entry = {
    "js/main": helpers.root("/src/main.ts"),
    "js/vendor-bundle": helpers.root("/src/vendor-bundle.ts"),
    "js/test": helpers.root("/src/test.ts")    
};

webpackConfig.output = {
    path: helpers.root("/dist"),
    filename: "[name].js"
};

webpackConfig.devtool = "source-map";

module.exports = webpackConfig;
```

**Extra info**:
I have attempted to look through for what is causing the compilation via karma to throw different errors than when webpack is called directly however I couldn't see any reason as the config files should be exactly the same from what I can tell? One thing to note is that adding a reference to the typings in the test.ts seemed to resolve the issue:
`/// <reference types="../typings" />`
This could also be done by adding typeroots / types pointing to the correct index.d.ts in the tsconfig compiler options.
Basically my bug report is that they behave differently - if anyone could shed any light as to why the import is needed I would be grateful as I have seen a number of people mention that definition imports change recently but I can't see anything in the typescript release notes?

**Comment by trusktr** [comment]
Slightly related, but I noticed that with `karma-webpack`, `Symbol` references are locally scoped, so it I assign to them, like `Symbol = 'something'`, this isn't actually writing to the global `Symbol`, so it doesn't work as expected. There's probably other unexpected differences too.

**Comment by codymikol** [comment]
Closing as this is 3 major versions old and I can't properly triage this. If you are still having issues with the current version 5.0.0, feel free to open another issue and I'll assist as best I can!
